app/controllers/jobseekers/sessions_controller.rb

```diff
@@ -11,8 +12,19 @@ class Jobseekers::SessionsController < Devise::SessionsController
     invalid not_found_in_database last_attempt
   ].map { |error| I18n.t("devise.failure.#{error}") }.freeze
 
+  def destroy
+    session.delete(:jobseeker_return_to)
+    super
+  end
+
   private
 
+  def destroy_jobseeker_return_to
+    return if request.referrer && (URI(request.referrer).path.include? "/job_application/new")
```

Comment:
Bear in mind not all browsers will set the referer header, so this wont' work for everyone.